hassio/src/dialogs/update/show-dialog-update.ts

```diff
@@ -6,13 +6,15 @@ export interface SupervisorDialogSupervisorUpdateParams {
   name: string;
   version: string;
   snapshotParams: any;
+  element: HTMLElement;
   updateHandler: () => Promise<void>;
 }
 
 export const showDialogSupervisorUpdate = (
   element: HTMLElement,
-  dialogParams: SupervisorDialogSupervisorUpdateParams
+  dialogParams: Partial<SupervisorDialogSupervisorUpdateParams>
 ): void => {
+  dialogParams.element = element;
```

Comment:
Make this a callback instead of an element to fire an event on

Reply:
Btw, we already have `updateHandler`, can't we do it in there?

Reply:
By moving it out, every place it's used will then be responsible, instead of having that responsibility on the dialog?
(I initially did that in the first commit)

Reply:
How many places are that?

Reply:
Now there are 2, when fully implemented 4

Reply:
Passing an element from every place vs firing an event on that element doesn't seem like a big burden? You could even make it a helper function that processes the info it gets from the `updateHandler` callback (name and version) and the element to fire the event on?